spyder/plugins/updatemanager/workers.py

```diff
@@ -118,28 +187,54 @@ def __init__(self, stable_only):
         super().__init__()
         self.stable_only = stable_only
         self.latest_release = None
-        self.releases = None
         self.update_available = False
         self.error = None
+        self.channel = None
 
-    def _check_update_available(self):
+    def _check_update_available(self, releases, github=True):
```

Comment:
```suggestion
    def _check_update_available(
        self,
        releases: list[packaging.version.Version],
        github: bool = True
    ):
```

I was a bit surprised to see `max` and `<` working for `releases` elements, so I think this makes things clearer.